cmd/kubeadm/app/apis/kubeadm/v1alpha1/zz_generated.deepcopy.go

```diff
@@ -48,12 +48,8 @@ func (in *AuditPolicyConfiguration) DeepCopyInto(out *AuditPolicyConfiguration)
 	*out = *in
 	if in.LogMaxAge != nil {
 		in, out := &in.LogMaxAge, &out.LogMaxAge
-		if *in == nil {
```

Comment:
Is this really what we want?  LogMaxAge is a pointer.  We're now blindly dereferencing `*in`.  This might be provable in some cases, but this changes ALL such pointer fields, it seems?

Reply:
`if in.LogMaxAge != nil` guards this.

Let me double check in the generator that we add these guards for every possible pointer case. And this should be tested.

Reply:
A pointer can be part of a struct, map, slice and pointed to by another pointer. All those cases add the guard in the generator:

- maps: https://github.com/kubernetes/gengo/blob/a6483bbdbc570bc82c882893c490f2c7e8d6bd0d/examples/deepcopy-gen/generators/deepcopy.go#L749
- slices: https://github.com/kubernetes/gengo/blob/a6483bbdbc570bc82c882893c490f2c7e8d6bd0d/examples/deepcopy-gen/generators/deepcopy.go#L784
- structs: https://github.com/kubernetes/gengo/blob/a6483bbdbc570bc82c882893c490f2c7e8d6bd0d/examples/deepcopy-gen/generators/deepcopy.go#L848
- pointers: https://github.com/kubernetes/gengo/blob/a6483bbdbc570bc82c882893c490f2c7e8d6bd0d/examples/deepcopy-gen/generators/deepcopy.go#L895

We have also test cases with fuzzing for all of those. So these cases should all be well tested.

Reply:
I think we still have some nil assignments which can be optimized (if the compiler does not do it).

Reply:
Ooops.  I think I saw that in the original generator change and just missed it here.  Taking another look.